run/jobs/main.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"time"
+)
+
+func getEnvVars() (string, string, string, string) {
```

Comment:
@grayside is this an ugly/awful solution?

Reply:
Try thinking about initializing a strongly typed config or app struct. Something you can createFromEnv or initialize by manually in tests. I can show you a couple examples of that pattern if helpful.

Reply:
How closely should it look like the canonical sample?  https://github.com/averikitsch/nodejs-docs-samples/blob/jobs/run/jobs/index.js

Reply:
Should we strongly type for the Job's defined variables, ie TASK_NUM and ATTEMPT_NUM? I lean on the side of idiomatic Go than matching the canonical.

Reply:
Go has some flexibility on this issue, and it depends.

In general, I prefer to have a layer that captures environment and "production" preferences such as: https://github.com/GoogleCloudPlatform/golang-samples/blob/main/getting-started/bookshelf/bookshelf.go#L56

A couple reasons for a central app struct:

1. If the app is organized "just so", it becomes easier for unit & integration tests to inject different config/behaviors without dealing with global state or adding checks deeper in the app.
2. Strongly typing/validating these values improve explicitness, local development, and portability. Problems can be detected on startup instead of deeper into the app's processing, and the requirements to run the app in a different environment than Jobs would be more clear.

Thanks for asking if we should follow canonical practice! +1 to "how to start the app follows language practice", but with new things how to draw that line can be be unclear.